src/main/java/io/tiledb/java/api/Array.java

```diff
@@ -483,7 +483,8 @@ public Pair getNonEmptyDomainFromIndex(long index) throws TileDBError {
     checkIsOpen();
     try (Domain domain = schema.getDomain();
         NativeArray domainArray =
-            new NativeArray(ctx, 2 * (int) domain.getRank(), domain.getType())) {
+            new NativeArray(
```

Comment:
This is not right, you don't need domain.rank() here. If the second parameter is "elements" then just 2 is enough.